nengo/dists.py

```diff
@@ -350,11 +373,53 @@ class SubvectorLength(SqrtBeta):
     --------
     SqrtBeta
     """
+
     def __init__(self, dimensions, subdimensions=1):
         super(SubvectorLength, self).__init__(
             dimensions - subdimensions, subdimensions)
 
 
+class CosineSimilarity(SubvectorLength):
+    """Distribution of dot products between random unit vectors.
```

Comment:
"Distribution of cosine similarity between two random vectors."
or maybe
"Distribution of the cosine of the angle between two random vectors."

then below: "The cosine similarity is given by the cosine of the angle between two vectors, which is equal to the norm of the dot product of the vectors, divided by the norms of the individual vectors".